### DISCUSSION TITLE
Issue creating custom rules plugin - require() of ES Module 'eslint-plugin-foo/dist/index.s' from '@eslint/eslintrc/dsist/eslintrc.cjs' not supported

### DISCUSSION THREAD

**Discussion by erikc96** [eslint/eslint#17067] [Q&A] [answered]
Hello everyone,

I'm trying to create a custom eslint plugin for my company, but I'm running into an issue that's preventing it from working. When I run eslint on Node 19.8.1 with eslint version 8.19.1, I get an error message that says my plugin is not supported:
`
Error [ERR_REQUIRE_ESM]: require() of ES Module /Users/BAZ/FOO/BitBucket/BAR/version-1/frontend/packages/eslint-plugin-bar/dist/index.js from /private/var/folders/xh/dy73h6r91n7ff132kfs9ppj40000gn/T/xfs-16bf8965/dlx-66312/node_modules/@eslint/eslintrc/dist/eslintrc.cjs not supported.
`

I'm transpiling the code for the eslint plugin to a minified esm bundle using esbuild, and I'm not sure if there are any good alternatives to this. Here's the pseudocode for my eslint configuration file:

`javascript
extends: [
    "plugin:bar/recommended",
],
plugins: ['bar'],
`
The custom eslint plugin is located in the packages subdirectory of the project workspace and is transpiled from a lib folder to a dist folder within the same plugin package.

I've checked that the \`eslint-plugin-bar\` package is installed and listed as a dependency in my \`package.json\` file. If anyone has any ideas on how to resolve this issue, I'd be very grateful for your help. Thanks in advance!

**Comment by chenxsan** [comment, marked as answer]
According to the [Node.js documentation](https://nodejs.org/api/esm.html#require), it is not possible to require an ES Module. However, you can bundle your eslint plugin as a CommonJS module using the esbuild API and its `format` option set to `cjs`, see https://esbuild.github.io/api/#format-commonjs

```sh
esbuild --format=cjs
```
